test/framework/debug.go

```diff
@@ -186,61 +186,10 @@ func DebugKube(cluster Cluster, mesh string, namespaces ...string) {
 
 	Logf("printing debug information of cluster %q for mesh %q and namespaces %q", cluster.Name(), mesh, namespaces)
 	for _, namespace := range namespaces {
-		nsDir := getNsDirPath(debugPath, cluster.Name(), namespace)
-		createDir(nsDir)
-
-		kubeOptions := *cluster.GetKubectlOptions(namespace) // copy to not override fields globally
-		kubeOptions.Logger = logger.Discard                  // to not print on stdout
-		out, err := k8s.RunKubectlAndGetOutputE(cluster.GetTesting(), &kubeOptions, "get", "all,kuma", "-oyaml")
-		if err != nil {
-			out = fmt.Sprintf("kubectl get for namespace %s failed with error: %s", namespace, err.Error())
+		nsErr := debugKubeNamespace(cluster, namespace, debugPath)
+		if nsErr {
 			errorSeen = true
 		}
-
-		// Ignore it if we don't have Gateway API resources installed
-		gatewayAPIOut, err := k8s.RunKubectlAndGetOutputE(cluster.GetTesting(), &kubeOptions, "get", "gateway-api", "-oyaml")
```

Comment:
Same as above, code here is extracted to function `debugKubeNamespace` underneath.